### PR TITLE
Fix: Do not allow nested nulls in keys at map construction.

### PR DESCRIPTION
**Pull request by laithsakka**
Summary:
Presto does not allow nested nulls in keys in map function
they are also changing map_from_entries to not allow nested
nulls in keys.

Adding the two checks to the Velox functions.

Differential Revision: D49287075

### CODE REVIEW

File: velox/common/base/VeloxException.h [hunk 1]
```diff
@@ -36,6 +36,9 @@ DECLARE_int32(velox_exception_system_stacktrace_rate_limit_ms);
 
 namespace facebook {
 namespace velox {
+static const char* kNullKeyErrorMessage = "map key cannot be null";
```
Comment (mbasmanova) [inline velox/common/base/VeloxException.h:39]: These are Presto-specific.... probably do not belong here.

File: velox/functions/prestosql/Map.cpp [hunk 2]
```diff
@@ -40,8 +40,22 @@ class MapFunction : public exec::VectorFunction {
...
         "Key and value arrays must be the same length";
-    static const char* kNullKey = "map key cannot be null";
-
+    auto checkNullsInKey =
+        [&](const auto& keysElements, auto offset, auto size) {
+          for (auto i = 0; i < size; ++i) {
+            if (keysElements->isNullAt(offset + i)) {
+              VELOX_USER_FAIL(kNullKeyErrorMessage);
+            }
+            if (keysElements->containsNullAt(offset + i)) {
```
Comment (mbasmanova) [inline velox/functions/prestosql/Map.cpp:49]: Why run this check twice?

File: velox/functions/prestosql/tests/MapFromEntriesTest.cpp [hunk 3]
```diff
@@ -396,3 +396,11 @@ TEST_F(MapFromEntriesTest, arrayOfConstantNotNulls) {
         {{{1, 2}}, {{1, 2}}, {{1, 2}}, {{1, 2}}});
   }
 }
+
+TEST_F(MapFromEntriesTest, nestedNullInKeys) {
+  VELOX_ASSERT_THROW(
+      evaluate(
+          "map_from_entries(array_constructor(row_constructor(array_constructor(null), null)))",
+          makeRowVector({makeFlatVector<int32_t>(1)})),
```
Comment (mbasmanova) [inline velox/functions/prestosql/tests/MapFromEntriesTest.cpp:404]: Let's update document to clarify this requirement.

File: velox/functions/prestosql/Map.cpp (hunk 2 above)
Comment (laithsakka) [reply velox/functions/prestosql/Map.cpp:49]: if the first fail the second is not reachable, 
do you suggest adding a new API containsNullAt that does not check the top level?

File: velox/common/base/VeloxException.h (hunk 1 above)
Comment (laithsakka) [reply velox/common/base/VeloxException.h:39]: I can move them

File: velox/functions/prestosql/tests/MapFromEntriesTest.cpp (hunk 3 above)
Comment (laithsakka) [reply velox/functions/prestosql/tests/MapFromEntriesTest.cpp:404]: ok

File: velox/functions/prestosql/MapFromEntries.cpp [hunk 4]
```diff
@@ -107,15 +113,24 @@ class MapFromEntriesFunction : public exec::VectorFunction {
...
             VELOX_USER_FAIL("map entry cannot be null");
           }
 
-          const bool isMapKeyNull =
-              keyValueVector->isNullAt(decodedValueVector->index(offset + i));
-          VELOX_USER_CHECK(!isMapKeyNull, "map key cannot be null");
+          // Check null keys.
+          auto keyIndex = decodedRowVector->index(offset + i);
+          if (keyVector->isNullAt(keyIndex)) {
+            VELOX_USER_FAIL(kNullKeyErrorMessage);
```
Comment (mbasmanova) [inline velox/functions/prestosql/MapFromEntries.cpp:125]: nit: Why not VELOX_USER_CHECK(!keyVector->isNullAt(keyIndex), kNullKeyErrorMessage) ?

**Review by mbasmanova — APPROVED** [review]
Thanks.

File: velox/functions/prestosql/MapFromEntries.cpp (hunk 4 above)
Comment (laithsakka) [reply velox/functions/prestosql/MapFromEntries.cpp:125]: will update before landing

**Pull request closed without merging**